kinto_http/tests/test_session.py

```diff
@@ -124,6 +124,15 @@ def test_raises_exception_if_headers_not_dict(session_setup: Tuple[MagicMock, Se
         session.request("get", "/test", headers=4)
 
 
+def test_get_request_with_data_raises_exception(session_setup: Tuple[MagicMock, Session]):
+    requests_mock, session = session_setup
+    response = get_200()
+    requests_mock.request.return_value = response
+
+    with pytest.raises(KintoException) as err:
+        session.request("GET", "/", data={"foo": "bar"})
+
+
```

Comment:
You can also add a test that fails with current master, which is:

```suggestion


def test_get_does_not_send_body(session_setup: Tuple[MagicMock, Session]):
    requests_mock, _ = session_setup
    response = get_200()
    requests_mock.request.return_value = response
    session = Session("https://example.org", timeout=4)
    assert session.auth is None
    session.request("GET", "/test")
    requests_mock.request.assert_called_with(
        "GET",
        "https://example.org/test",
        timeout=4,
        headers=requests_mock.request.headers,
    )

```